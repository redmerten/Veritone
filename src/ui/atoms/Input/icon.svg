<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16">
  <title>Eyeball</title>
  <path fill-rule="evenodd" d="M8 11a3 3 0 1 0 0-6 3 3 0 0 0 0 6zM0 8s2.997-5.008 8-5.008S16 8 16 8s-2.986 5.03-8 5.03C2.986 13.03 0 8 0 8zm7 0c0-.556.448-1 1-1 .556 0 1 .448 1 1 0 .556-.448 1-1 1-.556 0-1-.448-1-1zm0 0"/>
</svg>
