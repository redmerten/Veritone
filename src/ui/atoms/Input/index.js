// @flow

import { rem } from 'polished';
import styled, { css } from 'styled-components';

const Input = styled.input.attrs(({ dataTestid: dataTestId = 'atom-input', placeholder = '', type = 'text' }) => ({
  'data-testid': dataTestId,
  type,
  placeholder,
}))`
  appearance: textfield;
  background: none;
  border: 0;
  color: ${({ theme }) => theme.colors.raisinBlack};
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'initial')};
  font-family: ${({ theme }) => theme.fonts.roboto};
  font-size: ${rem(12)};
  font-weight: 400;
  margin: 0;
  outline: 0;
  padding: 0;
  width: inherit;

  &:-webkit-autofill {
    box-shadow: inset 0 0 0px 9999px ${({ theme }) => theme.colors.antiFlashWhite};
  }

  &::placeholder {
    color: ${({ theme }) => theme.colors.grayBlue};
  }

  &:disabled {
    color: ${({ theme }) => theme.colors.ashGrey} !important;
  }

  ${(p) =>
    p.large &&
    css`
      font-size: ${rem(16)};
      padding: ${rem(p.theme.spacing.md)} 0;
    `};
`;

Input.displayName = 'Input';

export default Input;
