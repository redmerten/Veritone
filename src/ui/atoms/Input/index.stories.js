// @flow

import { storiesOf } from '@storybook/react';
import React from 'react';

import Input from '.';

storiesOf('Atoms/Input', module)
  .add('default', () => <Input placeholder="Text input" type="text" />)
  .add('password', () => <Input placeholder="Password input" type="password" />);
