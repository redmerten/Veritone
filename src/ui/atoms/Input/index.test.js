// @flow

import React from 'react';

import Input from '.';

import { render } from 'utils/testing';

describe('atoms/Input', () => {
  it('matches its base snapshot', () => {
    const { baseElement } = render(<Input />);
    expect(baseElement).toMatchSnapshot();
  });

  it('renders input atom', () => {
    const { getByTestId } = render(<Input />);
    expect(getByTestId('atom-input')).toBeInTheDocument();
  });
});
