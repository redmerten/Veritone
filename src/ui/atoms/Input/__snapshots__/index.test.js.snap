// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Input matches its base snapshot 1`] = `
.c0 {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
  background: none;
  border: 0;
  color: #242425;
  cursor: initial;
  font-family: 'roboto-mono',monospace;
  font-size: 0.75rem;
  font-weight: 400;
  margin: 0;
  outline: 0;
  padding: 0;
  width: inherit;
}

.c0:-webkit-autofill {
  box-shadow: inset 0 0 0px 9999px #f2f3f5;
}

.c0::-webkit-input-placeholder {
  color: #8F95a5;
}

.c0::-moz-placeholder {
  color: #8F95a5;
}

.c0:-ms-input-placeholder {
  color: #8F95a5;
}

.c0::placeholder {
  color: #8F95a5;
}

.c0:disabled {
  color: #B0BEC5 !important;
}

<body>
  <div>
    <input
      class="c0"
      data-testid="atom-input"
      placeholder=""
      type="text"
    />
  </div>
</body>
`;
